main/lsp/requests/completion.cc

```diff
@@ -223,17 +223,28 @@ string methodSnippet(const core::GlobalState &gs, core::SymbolRef method) {
         }
     }
 
-    return fmt::format("{}({}){}", shortName, fmt::join(typeAndArgNames, ", "), "${0}");
+    if (typeAndArgNames.empty()) {
+        return fmt::format("{}{}", shortName, "${0}");
+    } else {
+        return fmt::format("{}({}){}", shortName, fmt::join(typeAndArgNames, ", "), "${0}");
+    }
 }
 
 unique_ptr<CompletionItem> getCompletionItemForKeyword(const LSPConfiguration &config, const RubyKeyword &rubyKeyword,
-                                                       size_t sortIdx) {
+                                                       string_view prefix, size_t sortIdx) {
     auto label = rubyKeyword.keyword;
     auto item = make_unique<CompletionItem>(label);
     item->sortText = fmt::format("{:06d}", sortIdx);
     item->kind = CompletionItemKind::Keyword;
+
+    // TODO(jez) Replace insertText with proper textEdit (replace the entire `fun`, to handle fuzzy prefixes)
```

Comment:
I'm going to follow up with this—the LSP spec takes liberty with how it uses the `insertText`:

```
The `insertText` is subject to interpretation by the client side.
Some tools might not take the string literally. For example
VS Code when code complete is requested in this example `con<cursor position>`
and a completion item with an `insertText` of `console` is provided it
will only insert `sole`. Therefore it is recommended to use `textEdit` instead
since it avoids additional client side interpretation.
```

So our hack in this file of removing the prefix is consistent with VS Code, but we should just not send insertText from LSP and instead send a textEdit (which has a range in the source, and the string to replace that range with). That requires tracking information in our AST about the `Loc` of the `fun` field of a Send.